homeassistant/components/gandi_livedns/config_flow.py

```diff
@@ -0,0 +1,127 @@
+"""Config flow for gandi_livedns."""
+from __future__ import annotations
+
+import logging
+
+from gandi_api_livedns import GandiApiLiveDNS
+from gandi_api_livedns.const import (
+    AVAILABLE_TYPE,
+    DEFAULT_IPV6,
+    DEFAULT_TIMEOUT,
+    DEFAULT_TTL,
+    DEFAULT_TYPE,
+)
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow
+from homeassistant.const import (
+    CONF_API_KEY,
+    CONF_DOMAIN,
+    CONF_NAME,
+    CONF_TIMEOUT,
+    CONF_TTL,
+    CONF_TYPE,
+)
+from homeassistant.helpers import config_validation as cv
+
+from .const import CONF_IPV6, CONF_UPDATE_INTERVAL, DEFAULT_UPDATE_INTERVAL, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_API_KEY): cv.string,
+        vol.Required(CONF_DOMAIN): cv.string,
+        vol.Required(CONF_NAME): cv.string,
+        vol.Optional(CONF_TYPE, default=DEFAULT_TYPE): vol.In(AVAILABLE_TYPE),
+        vol.Optional(CONF_TTL, default=DEFAULT_TTL): cv.positive_int,
+        vol.Optional(CONF_IPV6, default=DEFAULT_IPV6): cv.boolean,
+        vol.Optional(CONF_TIMEOUT, default=DEFAULT_TIMEOUT): cv.positive_int,
+        vol.Optional(
+            CONF_UPDATE_INTERVAL, default=DEFAULT_UPDATE_INTERVAL
+        ): cv.positive_int,
```

Comment:
So, we try to keep the initial setup/config flow, as compact as possible not to ask things that are actually options, and not part of the setup procedure.

Suggestions:
- "name", is confusing, as it is the rrname and not some casual integration instance name
- "timeout", does that have to be user-configurable even?
- "update_interval", does that have to be user-configurable? Can't we provide a sane default?
- "ttl","update_interval", "timeout" are all options, and (those that are kept) should go into a option flow instead.

Reply:
Indeed, the timeout is no longer necessary because it is managed on the module side (I'm the kind of people who give a maximum of control of user, but probably too many..)
after reflection i have changed the level of type, ttl and ipv6 from optional to required, these values is are mandatory for the update, i just took care to set good default values.
i have renamed name, ipv6, domain to better represent the data
I have kept the CONF_UPDATE_INTERVAL beacause the interval value depending on the ISP, and the rolling of ip address can be frequent.
I put 10 minutes by default (and I think it will be good for most cases ) and add a flow option ;) 
thank you for your patience